src/latch/ldata/_transfer/upload.py

```diff
@@ -337,26 +333,48 @@ def start_upload(
         chunk_size = chunk_size_mib * Units.MiB
 
     part_count = min(
-        latch_constants.maximum_upload_parts,
-        math.ceil(file_size / chunk_size),
+        latch_constants.maximum_upload_parts, math.ceil(file_size / chunk_size)
     )
     part_size = max(
-        chunk_size,
-        math.ceil(file_size / latch_constants.maximum_upload_parts),
+        chunk_size, math.ceil(file_size / latch_constants.maximum_upload_parts)
     )
 
     if throttle is not None:
         time.sleep(throttle.get_delay())
 
+    ingress_source: Optional[dict[str, str]] = None
+
+    try:
+        pod_id = Path("/root/.latch/id").read_text("utf-8")
+        ingress_source = {"pod_id": pod_id}
+    except FileNotFoundError:
+        sdk_token = user_config.token
+        if sdk_token != "":
+            ingress_source = {"sdk_token": sdk_token}
```

Comment:
same ting ... you might want to unify this code for getting the ingress / egress source.